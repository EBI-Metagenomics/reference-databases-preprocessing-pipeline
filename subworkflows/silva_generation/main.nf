
include { SILVA_REFORMAT } from '../../modules/silva_reformat/main.nf'
include { REMOVE_EMPTY_PHYLA } from '../../modules/remove_empty_phyla/main.nf'
include { MAKE_OTU_FILE } from '../../modules/make_otu_file/main.nf'
include { CLEAN_FASTA } from '../../modules/clean_fasta/main.nf'
include { GENERATE_MSCLUSTER } from '../../modules/generate_mscluster/main.nf'

workflow SILVA_GENERATION {

    take:
        subunit
        silva_taxdump
        download_fasta
        label
    main:

        dummy_fasta = file(params.dummy_fasta, checkIfExists: true)
        silva_fasta = file(download_fasta, checkIfExists: true)

        SILVA_REFORMAT(
            silva_fasta,
            silva_taxdump,
            subunit,
            params.silva_version,
            label
        )

        REMOVE_EMPTY_PHYLA(
            SILVA_REFORMAT.out.uplift,
            params.silva_version,
            label
        )

        MAKE_OTU_FILE(
            REMOVE_EMPTY_PHYLA.out.tax,
            SILVA_REFORMAT.out.taxid,
            params.silva_version,
            label
        )

        CLEAN_FASTA(
            SILVA_REFORMAT.out.fasta,
            REMOVE_EMPTY_PHYLA.out.tax,
            params.silva_version,
            label
        )

        // GENERATE_MSCLUSTER(
        //     dummy_fasta,
        //     CLEAN_FASTA.out.cleaned_fasta,
        //     REMOVE_EMPTY_PHYLA.out.tax,
        //     params.silva_version,
        //     label
        // )

    emit:
        fasta = CLEAN_FASTA.out.cleaned_fasta
        tax = REMOVE_EMPTY_PHYLA.out.tax
        otu = MAKE_OTU_FILE.out.otu

}