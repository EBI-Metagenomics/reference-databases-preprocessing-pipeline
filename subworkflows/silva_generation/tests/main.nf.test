nextflow_workflow {

    name "Test Workflow SILVA_GENERATION"
    script "subworkflows/silva_generation/main.nf"
    workflow "SILVA_GENERATION"
    tag "subworkflow"
    tag "silva_generation"

    test("SSU test") {
        tag "silva_generation_ssu"

        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = "SSU"
                input[1] = "${projectDir}/subworkflows/silva_generation/tests/data/taxdmp_2024-01-01.zip"
                input[2] = "${projectDir}/subworkflows/silva_generation/tests/data/test_ssu.fasta.gz"
                input[3] = "SILVA-SSU"
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.fasta).match("ssu_fasta") },
                { assert snapshot(workflow.out.tax).match("ssu_tax") },
                { assert snapshot(workflow.out.otu).match("ssu_otu") }
            )
        }

    }

    test("LSU test") {
        tag "silva_generation_lsu"

        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = "LSU"
                input[1] = "${projectDir}/subworkflows/silva_generation/tests/data/taxdmp_2024-01-01.zip"
                input[2] = "${projectDir}/subworkflows/silva_generation/tests/data/test_lsu.fasta.gz"
                input[3] = "SILVA-LSU"
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.fasta).match("lsu_fasta") },
                { assert snapshot(workflow.out.tax).match("lsu_tax") },
                { assert snapshot(workflow.out.otu).match("lsu_otu") }
            )
        }

    }

}
