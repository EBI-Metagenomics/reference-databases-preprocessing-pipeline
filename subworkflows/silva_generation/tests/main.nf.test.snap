{
    "ssu_otu": {
        "content": [
            [
                "SILVA-SSU.otu:md5,5e9204767114219849c18bbe2a1c7013"
            ]
        ],
        "timestamp": "2024-01-04T10:41:19.713285359"
    },
    "lsu_tax": {
        "content": [
            [
                "SILVA-LSU-tax.txt:md5,792ff27421ffe3152b3cdea6c48dd8f6"
            ]
        ],
        "timestamp": "2024-01-04T10:44:28.822536006"
    },
    "lsu_fasta": {
        "content": [
            [
                "SILVA-LSU.fasta:md5,efe4a0ee53c4620bbc365aacb351d828"
            ]
        ],
        "timestamp": "2024-01-04T10:44:28.724528153"
    },
    "lsu_otu": {
        "content": [
            [
                "SILVA-LSU.otu:md5,667ba8ac67bc84dc60936cd4d34490ce"
            ]
        ],
        "timestamp": "2024-01-04T10:44:28.870885353"
    },
    "ssu_tax": {
        "content": [
            [
                "SILVA-SSU-tax.txt:md5,f7e95c1b79ed6034b1a04e52c6355486"
            ]
        ],
        "timestamp": "2024-01-04T10:41:19.689362938"
    },
    "ssu_fasta": {
        "content": [
            [
                "SILVA-SSU.fasta:md5,48d03de1ea1b1c38ffeec0e00b0061ff"
            ]
        ],
        "timestamp": "2024-01-04T10:41:19.614252074"
    }
}